editor/plugins/gdextension/cpp_scons/cpp_scons_gdext_creator.cpp

```diff
@@ -0,0 +1,232 @@
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#include "cpp_scons_gdext_creator.h"
+
+#include "core/core_bind.h"
+#include "core/io/dir_access.h"
+#include "core/version.h"
+#include "gdextension_template_files.gen.h"
+
+#include "editor/editor_node.h"
+
+void CppSconsGDExtensionCreator::_git_clone_godot_cpp(const String &p_parent_path, bool p_compile) {
+	EditorProgress ep("Preparing GDExtension C++ plugin", "Preparing GDExtension C++ plugin", 3);
+	ep.step("Checking for Git...", 0);
+	List<String> args;
+	args.push_back("--version");
+	String output = "";
+	int result = OS::get_singleton()->execute("git", args, &output);
+	ERR_FAIL_COND_MSG(result != 0 || output.is_empty(), "Could not run git command. Please clone godot-cpp manually in order to have a working GDExtension plugin.");
+	args.front()->set("clone");
+	args.push_back("--single-branch");
+	args.push_back("--branch");
+	args.push_back(VERSION_BRANCH);
+	args.push_back("https://github.com/godotengine/godot-cpp");
+	const String godot_cpp_path = p_parent_path.trim_prefix("res://").path_join("godot-cpp");
+	args.push_back(godot_cpp_path);
+	ep.step("Cloning godot-cpp...", 1);
+	output = "";
+	result = OS::get_singleton()->execute("git", args, &output);
+	Ref<DirAccess> dir = DirAccess::create(DirAccess::ACCESS_RESOURCES);
+	if (result != 0 || !dir->dir_exists(godot_cpp_path)) {
+		args.get(3) = "master";
+		output = "";
+		result = OS::get_singleton()->execute("git", args, &output);
+	}
+	ERR_FAIL_COND_MSG(result != 0 || !dir->dir_exists(godot_cpp_path), "Failed to clone godot-cpp. Please clone godot-cpp manually in order to have a working GDExtension plugin.");
+	ep.step("Performing initial compile... (this may take several minutes)", 2);
+	if (p_compile) {
+		result = OS::get_singleton()->execute("scons", List<String>());
+		ERR_FAIL_COND_MSG(result != 0, "Failed to compile godot-cpp. Please ensure SCons is installed, then run the `scons` command in your project.");
+	}
+	ep.step("Done!", 3);
+}
+
+void CppSconsGDExtensionCreator::_make_dir_in_res(const String &p_dir_path) {
```

Comment:
Why not use the static `make_dir_recursive_absolute()`? Though not sure if it works with `res://` paths (but they are technically absolute, in project scope).

Reply:
Good idea, that works.